src/main/java/randoop/generation/DemandDrivenInputCreation.java

```diff
@@ -255,8 +255,10 @@ private static Set<TypedOperation> producerMethodSearch(Type t, Type startingTyp
         List<Executable> executableList = new ArrayList<>();
 
         // Adding constructors if the current type is what we are looking for.
-        if (t.isAssignableFrom(currentType)) {
-          Collections.addAll(executableList, currentClass.getConstructors());
+        if (t.isAssignableFrom(currentType) && !Modifier.isAbstract(currentClass.getModifiers())) {
+          for (Constructor<?> constructor : currentClass.getConstructors()) {
```

Comment:
I understand the addition of the test on the previous line.
What is the point of expanding a call to `addAll()` into a `for` loop?

Reply:
I just fixed it. I am not sure how that got in there.